android/src/main/java/com/clevertap/react/CleverTapListenerProxy.kt

```diff
@@ -0,0 +1,224 @@
+package com.clevertap.react
+
+import android.util.Log
+import com.clevertap.android.sdk.CTFeatureFlagsListener
+import com.clevertap.android.sdk.CTInboxListener
+import com.clevertap.android.sdk.CleverTapAPI
+import com.clevertap.android.sdk.InAppNotificationButtonListener
+import com.clevertap.android.sdk.InAppNotificationListener
+import com.clevertap.android.sdk.InboxMessageButtonListener
+import com.clevertap.android.sdk.InboxMessageListener
+import com.clevertap.android.sdk.PushPermissionResponseListener
+import com.clevertap.android.sdk.SyncListener
+import com.clevertap.android.sdk.displayunits.DisplayUnitListener
+import com.clevertap.android.sdk.displayunits.model.CleverTapDisplayUnit
+import com.clevertap.android.sdk.inapp.CTInAppNotification
+import com.clevertap.android.sdk.inbox.CTInboxMessage
+import com.clevertap.android.sdk.product_config.CTProductConfigListener
+import com.clevertap.android.sdk.pushnotification.CTPushNotificationListener
+import com.facebook.react.bridge.Arguments
+import org.json.JSONException
+import org.json.JSONObject
+
+object CleverTapListenerProxy : SyncListener, InAppNotificationListener, CTInboxListener,
+    InboxMessageButtonListener, InboxMessageListener, InAppNotificationButtonListener,
+    DisplayUnitListener, CTProductConfigListener, CTFeatureFlagsListener,
+    CTPushNotificationListener, PushPermissionResponseListener {
+
+    private const val LOG_TAG = Constants.REACT_MODULE_NAME
+
+    fun attachToInstance(instance: CleverTapAPI) {
+        instance.registerPushPermissionNotificationResponseListener(this)
+        instance.ctPushNotificationListener = this
```

Comment:
@vasct 

`instance.registerPushPermissionNotificationResponseListener` adds a listener to a list and doesn't replace it https://github.com/CleverTap/clevertap-android-sdk/blob/4b1ea7577e14f9abb68492ad09b36cf01c3b7c48/clevertap-core/src/main/java/com/clevertap/android/sdk/CallbackManager.java#L195

`attachToInstance` is called twice, once from `CleverTapApplication` and once from `ClevertapModuleImpl` which adds 2 listeners to the list and hence this callback is received twice on the js side

Reply:
Good catch, I will fix it.